### PR TITLE
Introduce multi-layered canvas rendering for optimization

### PR DESCRIPTION
**Pull request by DevSDK**
Previously, The all blocks are rendered every events or updates.
It could possibly cause performance issue.
Please see [1]. The keyboard event spends 20ms so the rendering update is delayed which over 16ms for 60 FPS.
It may cause a bad-user experience.

Therefore, Separate rendering layers to reduce rendering time.
The background tiles are rendered only once when the canvas is initialized.
Also, Stacked blocks are not required to render  every time. So add "shouldUpdateStackedLayer" to use checking update the stacked block.

So, update blocks are only control-blocks.

Note here [2], the rendering time is reduced to 13~6ms.

Also, this CL contains minor refactoring for draw-rects

[1] https://user-images.githubusercontent.com/18409763/110893398-40e42480-8339-11eb-928e-fb41a4ca7020.png

[2] https://user-images.githubusercontent.com/18409763/110893887-34140080-833a-11eb-878f-7156b6fb81e7.png

### CODE REVIEW

**Comment by WoochanLee** [comment]
Thank you for contributing!!
I'll do a code review soon.
Could you please change your PR base branch to 'develop'?
I created develop branch for contributing a few days ago.
Anyway, very Thanks!

**Comment by DevSDK** [comment]
> Could you please change your PR base branch to 'develop'?

Done.

**Comment by WoochanLee** [comment]
@oh3vci Please review this code as well.

File: components/game/screen.js
Comment (WoochanLee) [inline components/game/screen.js:31]: What is this value's purpose?
I couldn't find the location that makes this value false except here.

Comment (DevSDK) [reply components/game/screen.js:31]: Oh I see, It was mistaken.
The expected work is when the flag is true, this should new update render the stacked blocks.
Therefore, I've add clear block function which used clear marked blocks which is marked as previously rendered block.

**Comment by DevSDK** [comment]
PTAL

File: components/game/screen.js
Comment (DevSDK) [inline components/game/screen.js:171]: I've add the 'false' to block not-required repeated rendering.

File: components/game/screen.js
Comment (DevSDK) [inline components/game/screen.js:134]: This is necessary to wipe previously rendered block. It might be needed refactoring.

**Comment by WoochanLee** [comment]
Please check this bug.
An afterimage remains when using the shift function.
스크린샷 2021-03-20 오후 5 08 10

**Comment by DevSDK** [comment]
PTAL

**Review by WoochanLee — APPROVED** [review]

**Comment by oh3vci** [comment]
LGTM!

**Comment by WoochanLee** [comment]
Don't merge before has two approved.
I will change branch rules.

**Pull request merged**
